tests/src/tests/command_test_utils.go

```diff
@@ -0,0 +1,256 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/yaml"
+
+	schema "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+)
+
+func AddEnv(numEnv int) []schema.EnvVar {
+	commandEnvs := make([]schema.EnvVar, numEnv)
+	for i := 0; i < numEnv; i++ {
+		commandEnvs[i].Name = "Name_" + GetRandomString(5, false)
+		commandEnvs[i].Value = "Value_" + GetRandomString(5, false)
+		LogMessage(fmt.Sprintf("   ....... Add Env: %s", commandEnvs[i]))
+	}
+	return commandEnvs
+}
+
+func AddAttributes(numAtrributes int) map[string]string {
+	attributes := make(map[string]string)
+	for i := 0; i < numAtrributes; i++ {
+		AttributeName := "Name_" + GetRandomString(6, false)
+		attributes[AttributeName] = "Value_" + GetRandomString(6, false)
+		LogMessage(fmt.Sprintf("   ....... Add attribute : %s = %s", AttributeName, attributes[AttributeName]))
+	}
+	return attributes
+}
+
+func addGroup() *schema.CommandGroup {
+
+	commandGroup := schema.CommandGroup{}
+
+	commandGroup.Kind = GetRandomGroupKind()
+	LogMessage(fmt.Sprintf("   ....... group Kind: %s", commandGroup.Kind))
+	commandGroup.IsDefault = GetBinaryDecision()
```

Comment:
this might generate an invalid devfile, since each command group can only have one default command. 
check the point 5 in commands validation: https://github.com/devfile/api/blob/master/pkg/validation/validation-rule.md

```
{build, run, test, debug}, each kind of group can only have one default command associated with it. If there are multiple commands of the same kind without a default, a warning will be displayed.
```

Reply:
At the moment the code only adds one group to any single command, so this is not currently a problem. I may update the code to add multiple in which case I can handle the requirement then.

Reply:
Sounds good. might want to open an issue to track for multiple command tests